creme/time_series/exponential_smoothing.py

```diff
@@ -0,0 +1,566 @@
+    Level equation
+    .. math:: \ell_{t}=\alpha y_{t}+(1-\alpha)\left(\ell_{t-1}+\phi b_{t-1}\right)
+    Trend equation
+    .. math:: b_{t}=\beta^{*}\left(\ell_{t}-\ell_{t-1}\right)+\left(1-\beta^{*}\right) \phi b_{t-1}
+
+    Parameters:
+        alpha (float): The smoothing parameter for the level, 0 ≤ alpha ≤ 1 .Defaults to `0.5`.
+        beta (float): The smoothing parameter for the trend, 0 ≤ beta ≤ 1 .Defaults to `0.5`.
+        phi (float): damping parameter  0 < phi < 1. Defaults to `0.8`.
+        l0 (float): Initialization value for the level. Defaults to `0`.
+        b0 (float): Initialization value for the trend. Defaults to `0`.
+
+    Example:
+    ::
+        >>> #TODO
+
+    References:
+        1. `Damped trend methods <https://otexts.com/fpp2/holt.html>`_
+    """
+
+    def __init__(self,
+                 alpha: float = 0.5,
+                 beta: float = 0.5,
+                 phi: float = 0.8,
+                 l0: float = 0,
+                 b0: float = 0):
+
+        self.alpha = alpha
+        self.beta = beta
+        self.phi = phi
+
+        self.lt = l0
+        self.bt = b0
+
+    def fit_one(self, y):
+
+        lt_1 = self.lt
+        bt_1 = self.bt
+
+        self.lt = self.alpha * y + (1 - self.alpha) * (lt_1 + self.phi * bt_1)
+        self.bt = self.beta * (self.lt - lt_1) + (
+            1 - self.beta) * bt_1 * self.phi
+
+        return self
+
+    def forecast(self, horizon: int) -> list:
+        return [
+            self.lt + (_discount_sum(self.phi, h) * self.bt)
+            for h in range(1, horizon + 1)
+        ]
+
+
+class HoltWinterAdditive(base.Forecaster):
+    """
+    Holt (1957) and Winters (1960) extended Holt’s method to capture seasonality.
+    The Holt-Winters seasonal method comprises the forecast equation
+    and three smoothing equations — one for the level  $l_{t}$,
+    one for the trend $b_{t}$ , and one for the seasonal component $s_{t}$,
+    with corresponding smoothing parameters alpha, beta and gamma.
+    We use m to denote the frequency of the seasonality, 
+    i.e., the number of seasons in a year. For example, for quarterly data  m=4, and for monthly data  m=12.
+
+    Parameters:
+        m (int): m the frequency of the seasonality
+        alpha (float): The smoothing parameter for the level, 0 ≤ alpha ≤ 1 . Defaults to `0.5`.
+        beta (float): The smoothing parameter for the trend, 0 ≤ beta ≤ 1 . Defaults to `0.5`.
+        gamma (float): The smoothing parameter seasonal component 0 ≤ gamma ≤ 1. Defaults to `0.5`.
+        s (List[float]): Initialization values for the seasonality. Defaults to `None`.
+        l0 (float): Initialization value for the level. Defaults to `0`.
+        b0 (float): Initialization value for the trend. Defaults to `0`.
+
+    Example:
+    ::
+        >>> #TODO
+
+    References:
+        1. `Holt-Winters’ additive method <https://otexts.com/fpp2/holt-winters.html>`_
+    """
+
+    def __init__(
```

Comment:
In `creme` we've mostly adopted a style of writing `__init__`'s in a single line, and line wrapping after 100 characters. Not really important though.